src/main/resources/com/google/api/codegen/js/sample.snip

```diff
@@ -0,0 +1,181 @@
+        });
+      @else
+        {@class.requestVarName}.then(function() {}, function(reason) {
+          console.log('error: ' + reason.result.error.message');
+        });
+      @end
+    @end
+  }
+
+  @if class.isPageStreaming
+    function {@pageStreamingFuncSignature(class)} {
+      var {@class.requestVarName} = {@requestFuncSignature(class)};
+      request.then(function({@class.responseVarName}) {
+        var {@class.pageStreaming.pageVarName} = {@class.responseVarName}.result.{@class.pageStreaming.resourceFieldName};
+        @if class.pageStreaming.isResourceRepeated
+          @if class.pageStreaming.isResourceMap
+            Object.keys({@class.pageStreaming.pageVarName}).forEach(function({@class.pageStreaming.resourceKeyVarName}) {
+              // TODO: Change code below to process each `{@class.pageStreaming.resourceKeyVarName}` property:
+              console.log({@class.pageStreaming.resourceKeyVarName} + ': ' + JSON.stringify({@class.pageStreaming.pageVarName}[{@class.pageStreaming.resourceKeyVarName}], null, 2));
+            });
+          @else
+            for (var i = 0; i < {@class.pageStreaming.pageVarName}.length; i++) {
+              // TODO: Change code below to process each resource in `{@class.pageStreaming.pageVarName}`:
+              console.log(JSON.stringify({@class.pageStreaming.pageVarName}[i], null, 2));
+            }
+          @end
+        @else
+          // TODO: Change code below to process each `{@class.pageStreaming.pageVarName}` resource:
+          console.log({@class.pageStreaming.pageVarName});
+        @end
+
+        if ({@class.responseVarName}.nextPageToken) {
+          @if class.pageStreaming.isResourceSetterInRequestBody
+            {@class.requestBodyVarName}.pageToken = {@class.responseVarName()}.nextPageToken;
+          @else
+            {@class.paramsVarName}.pageToken = {@class.responseVarName()}.nextPageToken;
+          @end
+          {@pageStreamingFuncSignature(class)};
+        }
+      }, function(reason) {
+        console.log('error: ' + reason.result.error.message);
+      });
+    }
+
+  @end
+  function start() {
+    gapi.client.init({
+      @switch class.auth.type
+      @case "API_KEY"
+        'apiKey': '',  // TODO: Update placeholder with desired API key.'
+      @default
+        'clientId': '',  // TODO: Update placeholder with desired client ID.
+        'scope': '{@scopeString(class.auth.scopes)}',
+      @end
+      'discoveryDocs': ['https://content.googleapis.com/discovery/v1/apis/{@class.apiName}/{@class.apiVersion}/rest'],
+    }).then(function() {
+      @if class.auth.type != "API_KEY"
+        gapi.auth2.getAuthInstance().signIn();
+        gapi.auth2.getAuthInstance().isSignedIn.listen(function(isSignedIn) {
+          if (isSignedIn) {
+            makeApiCall();
+          }
+        });
+      @else
+        makeApiCall();
+      @end
+    });
+  }
+  gapi.load('client', start);
+  </script>
+@end
+
+@private requestFuncSignature(class)
```

Comment:
What's the purpose of the second argument to `argList`? It appears to be always `@FALSE`, and I'm confused by the doc string.